lib/install/deps.js

```diff
@@ -74,7 +74,7 @@ function doesChildVersionMatch (child, requested, requestor) {
     return false
   }
   try {
-    return semver.satisfies(child.package.version, requested.fetchSpec)
+    return semver.satisfies(child.package.version, requested.fetchSpec, true)
```

Comment:
It is actually documented and you find multiple descriptions as e.g.:

```All methods and classes take a final loose boolean argument that, if true, will be more forgiving about not-quite-valid semver strings. The resulting output will always be 100% strict, of course.```

I personally think it would be a good thing to just go ahead and document this breaking change instead of loosening the strictness but that's just me. It could of course also be stricter in a future major, I think it would still be fine to just document it now as breaking changes are expected for 5.0 anyway.